python/pyarrow/includes/libarrow_dataset.pxd

```diff
@@ -300,10 +300,6 @@ cdef extern from "arrow/dataset/api.h" namespace "arrow::dataset" nogil:
     cdef cppclass CPartitioningFactory "arrow::dataset::PartitioningFactory":
         pass
 
-    cdef cppclass CDefaultPartitioning \
-            "arrow::dataset::DefaultPartitioning"(CPartitioning):
-        CDefaultPartitioning()
-
```

Comment:
You can probably remove `InMemoryScanTask` from this file as well

Reply:
Indeed

Reply:
Removed